packages/devtools_app/test/flutter/table_test.dart

```diff
@@ -511,6 +513,49 @@ void main() {
       expect(tree.children[2].name, equals('Baz'));
     });
 
+    testWidgets('selection changes with arrow keys',
+        (WidgetTester tester) async {
+      final data = TestData('Foo', 0)
+        ..children.addAll([
+          TestData('Bar', 1),
+          TestData('Crackle', 5),
+        ])
+        ..expandCascading();
+      final table = TreeTable<TestData>(
+        columns: [
+          _NumberColumn(),
+          treeColumn,
+        ],
+        dataRoots: [data],
+        treeColumn: treeColumn,
+        keyFactory: (d) => Key(d.name),
+        sortColumn: treeColumn,
+        sortDirection: SortDirection.ascending,
+      );
+
+      await tester.pumpWidget(wrap(table));
+      await tester.pumpAndSettle();
+
+      final fooFinder = find.byKey(const Key('Foo'));
+
+      await tester.tap(fooFinder);
+      await tester.pumpAndSettle();
+      final TreeTableState state = tester.state(find.byWidget(table));
+
+      expect(state.selectedNode, equals(data.root));
+
+      await tester.sendKeyDownEvent(LogicalKeyboardKey.arrowDown);
```

Comment:
this test isn't passing yet. `tester.sendKeyDownEvent` isn't triggering my callback but I don't know why.

Reply:
have you tried `sendKeyEvent` instead of `sendKeyDownEvent`?

Reply:
that did give me a KeyUpEvent... but no keydowns

Reply:
try specifying a platform.

`await tester.sendKeyEvent(LogicalKeyboardKey.arrowDown, platform: 'macos');`

In the flame chart test that tests key stroke see this comment: "Use platform macos so that we have access to [event.data.keyLabel]. Event simulation is not supported for platform 'web'."

Reply:
Fixed now. I'm testing logicalKey which seems to work without specifying a platform. The main problem I had was that focus wasn't being applied